src/WinGetServer/WinGetServerManualActivation_Client.cpp

```diff
@@ -0,0 +1,152 @@
+    THROW_HR_IF(HRESULT_FROM_WIN32(status), status != RPC_S_OK);
+    bindingPtr.reset(&binding);
+
+    status = RpcBindingFromStringBindingA(binding, &WinGetServerManualActivation_IfHandle);
+    THROW_HR_IF(HRESULT_FROM_WIN32(status), status != RPC_S_OK);
+}
+
+HRESULT LaunchWinGetServerWithManualActivation()
+{
+    const std::filesystem::path& localAppDataPath = GetKnownFolderPath(FOLDERID_LocalAppData);
+    std::wstring serverExePath = std::wstring{ localAppDataPath } + std::wstring{ s_ServerExePath } + L" --manualActivation";
+
+    STARTUPINFO info = { sizeof(info) };
+    PROCESS_INFORMATION processInfo;
+    if (CreateProcessW(NULL, &serverExePath[0], NULL, NULL, FALSE, 0, NULL, NULL, &info, &processInfo))
+    {
+        WaitForSingleObject(processInfo.hThread, 500);
+        CloseHandle(processInfo.hProcess);
+        CloseHandle(processInfo.hThread);
+        return S_OK;
+    }
+    else
+    {
+        return HRESULT_FROM_WIN32(GetLastError());
+    }
+}
+
+HRESULT CallCreateInstance(const CLSID* clsid, const IID* iid, UINT32 flags, UINT32* bufferByteCount, BYTE** buffer)
+{
+    RpcTryExcept
+    {
+        RETURN_IF_FAILED(CreateInstance(*clsid, *iid, flags, bufferByteCount, buffer));
+    }
+        RpcExcept(1)
+    {
+        return HRESULT_FROM_WIN32(RpcExceptionCode());
+    }
+    RpcEndExcept;
+
+    return S_OK;
+}
+
+HRESULT LaunchServerAndCreateInstance(const CLSID* clsid, const IID* iid, UINT32 flags, void** out)
+{
+    UINT32 bufferByteCount = 0;
+    BYTE* buffer = nullptr;
+    UniqueMidl bufferPtr;
+
+    RETURN_IF_FAILED(LaunchWinGetServerWithManualActivation());
+    RETURN_IF_FAILED(CallCreateInstance(clsid, iid, flags, &bufferByteCount, &buffer));
+
+    bufferPtr.reset(buffer);
+
+    wil::com_ptr<IStream> stream;
+    RETURN_IF_FAILED(CreateStreamOnHGlobal(nullptr, TRUE, &stream));
+    RETURN_IF_FAILED(stream->Write(buffer, bufferByteCount, nullptr));
+    RETURN_IF_FAILED(stream->Seek({}, STREAM_SEEK_SET, nullptr));
+
+    wil::com_ptr<IUnknown> output;
+    RETURN_IF_FAILED(CoUnmarshalInterface(stream.get(), *iid, reinterpret_cast<void**>(&output)));
+    *out = output.detach();
+    return S_OK;
+}
+
+extern "C" HRESULT WinGetServerManualActivation_CreateInstance(const CLSID* clsid, const IID* iid, UINT32 flags, void** out)
+{
+    RETURN_HR_IF_NULL(E_POINTER, clsid);
+    RETURN_HR_IF_NULL(E_POINTER, iid);
+    RETURN_HR_IF_NULL(E_POINTER, out);
+
+    if (WinGetServerManualActivation_IfHandle == NULL)
+    {
+        static std::once_flag rpcBindingOnce;
+        try
+        {
+            std::call_once(rpcBindingOnce, InitializeRpcBinding);
+        }
+        CATCH_RETURN();
+    }
+
+    HRESULT result;
+
+    for (int i = 0; i < 3; i++)
+    {
+        result = LaunchServerAndCreateInstance(clsid, iid, flags, out);
+        if (SUCCEEDED(result) || result == HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND))
+        {
+            break;
+        }
+    }

```

Comment:
Put a short sleep in between the loop iterations when the manual reset event code is in place, otherwise we will try all 3 times almost instantly.  We want enough time to elapse to cover the case of the server shutting down while we are trying to create an object.

Reply:
Added a short sleep at the end of each loop iteration